application/src/test/java/com/yahoo/application/ApplicationTest.java

```diff
@@ -262,6 +265,25 @@ void component() throws Exception {
         }
     }
 
+    @Test
+    void application_generation_metric() throws Exception {
+        try (ApplicationFacade app = new ApplicationFacade(Application.fromBuilder(new Application.Builder().container("default", new Application.Builder.Container()
+                                                                                                                                          .component(ComponentWithMetrics.class))))) {
+            var component = (ComponentWithMetrics)app.getComponentById(ComponentWithMetrics.class.getName());
+            assertNotNull(component);
+            var metrics = (SimpleMetricConsumer)component.metrics().newInstance(); // not actually a new instance
+            assertNotNull(metrics);
+            int maxWaitMs = 10000;
+            Bucket snapshot = null;
+            while (maxWaitMs-- > 0 && ( snapshot = metrics.receiver().getSnapshot() ) == null) {
+                Thread.sleep(1);
+            }
+            assertNotNull(snapshot);
+            assertEquals(1, snapshot.getValuesForMetric("application_generation").size());
+            assertEquals(0, snapshot.getValuesForMetric("application_generation").iterator().next().getValue().getLast());
```

Comment:
Out of curiosity, where did the test fail?

Reply:
On 282 - it never got the metric. 

In a real setup we still never get the metric any more though, so this didn't solve the problem.